src/main/java/com/epam/javacourses2016/task12/SolverTask12.java

```diff
@@ -1,22 +1,40 @@
 package com.epam.javacourses2016.task12;
 
+import java.util.Collections;
 import java.util.List;
 
 /**
  * Задан список целых чисел и число X.
  * Не используя вспомогательных объектов и не изменяя размера списка,
- *    переставить элементы списка так, чтобы сначала шли числа, не превосходящие X, а затем числа, большие X.
+ * переставить элементы списка так, чтобы сначала шли числа, не превосходящие X, а затем числа, большие X.
  */
 public class SolverTask12 {
 
     /**
      * Преобразует целочисленный список таким образом, чтобы сначала шли числа меньшие value, затем большие.
+     *
      * @param integers Целочисленный список.
-     * @param value Разделительное значение.
+     * @param value    Разделительное значение.
      * @return Преобразованный список.
      */
     List<Integer> transform(List<Integer> integers, int value) {
-        //TODO
-        return null;
+
+        int indexStart = integers.size() - 1;
+
+        for (int i = 0; i < integers.size(); i++) {
```

Comment:
Ok. I got it. And what about new Array List with initial capacity, it will be better than LinkedList? 
Smth like this:
```
        List<Integer> arrayList = new ArrayList<>(integers.size());
        int beginIndex = 0;
        int endIndex = integers.size() - 1;

        for (int i = 0; i < integers.size(); i++) {
            int listValue = integers.get(i);
            if (listValue > value) {
                arrayList.add(endIndex, listValue);
                endIndex--;
            } else {
                arrayList.add(beginIndex, listValue);
                beginIndex++;
            }
        }
```